ntoskrnl/inbv/inbv.c

```diff
@@ -210,6 +226,111 @@ BootLogoFadeIn(VOID)
     }
 }
 
+static VOID
+RemovePalette(IN PVOID Image)
+{
+    PBITMAPINFOHEADER BitmapInfoHeader;
+    LPRGBQUAD Palette;
+
+    /* Get bitmap header and palette */
+    BitmapInfoHeader = Image;
+    Palette = (LPRGBQUAD)((PUCHAR)Image + BitmapInfoHeader->biSize);
+
+    /* Remove palette information */
+    RtlZeroMemory(Palette, sizeof(MainPalette));
```

Comment:
This causes a BSOD on MSVC-compiled NTOSKRNL.EXE, because it has the effect of writing into read-only memory.

Reply:
Fixed by explicitly changing the memory protection of the kernel resource section. This is OK, a similar operation is also done during bugcheck. This is needed because we are indeed modifying the contents of the resources (here, patching the palette of bitmap resources).